x-pack/plugins/fleet/server/services/spaces/helper.test.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Settings } from '../../types';
+import { appContextService } from '../app_context';
+import { getSettingsOrUndefined } from '../settings';
+
+import { isSpaceAwarenessEnabled, isSpaceAwarenessMigrationPending } from './helpers';
+
+jest.mock('../app_context');
+jest.mock('../settings');
+
+function mockFeatureFlag(val: boolean) {
+  jest.mocked(appContextService.getExperimentalFeatures).mockReturnValue({
+    useSpaceAwareness: val,
+  } as any);
+}
+
+function mockGetSettings(settings?: Partial<Settings>) {
+  if (settings) {
+    jest.mocked(getSettingsOrUndefined).mockResolvedValue(settings as any);
+  } else {
+    jest.mocked(getSettingsOrUndefined).mockResolvedValue(undefined);
+  }
+}
+
+describe('isSpaceAwarenessEnabled', () => {
+  beforeEach(() => {
+    jest.mocked(appContextService.getExperimentalFeatures).mockReset();
+    jest.mocked(getSettingsOrUndefined).mockReset();
+  });
+  it('should return false if feature flag is disabled', async () => {
+    mockFeatureFlag(false);
+    const res = await isSpaceAwarenessEnabled();
+
+    expect(res).toBe(false);
```

Comment:
nit, just thought I'd mention this jest helper:
```suggestion
    await expect(isSpaceAwarenessEnabled()).resolves.toBe(false);
```